launcher/ui/pages/instance/LogPage.cpp

```diff
@@ -36,331 +36,288 @@
-    }
-    else
-    {
-        ui->trackLogCheckbox->setCheckState(Qt::Checked);
-    }
+    saveSettings();
+
+    return true;
 }
 
-void LogPage::UIToModelState()
+void LogPage::onStatusChanged(QQuickWidget::Status status)
 {
-    if(!m_model)
-    {
-        return;
-    }
-    m_model->setLineWrap(ui->wrapCheckbox->checkState() == Qt::Checked);
-    m_model->suspend(ui->trackLogCheckbox->checkState() != Qt::Checked);
+    if (status == QQuickWidget::Status::Ready)
+        onQuickWidgetReady();
+}
+
+void LogPage::onQuickWidgetReady()
+{
+    [[maybe_unused]] bool no_errors = QQmlProperty::write(rootObject(), "logModel", QVariant::fromValue(m_proxy));
+    Q_ASSERT(no_errors);
+
+    no_errors = connect(rootObject(), SIGNAL(onWrapModeChanged(int)), this, SLOT(wrapModeChanged(int)));
+    Q_ASSERT(no_errors);
+    no_errors = connect(rootObject(), SIGNAL(onSuspendedChanged(bool)), this, SLOT(suspendedChanged(bool)));
+    Q_ASSERT(no_errors);
+    no_errors = connect(rootObject(), SIGNAL(onUseRegexChanged(bool)), this, SLOT(useRegexChanged(bool)));
+    Q_ASSERT(no_errors);
+    no_errors = connect(rootObject(), SIGNAL(onCopyPressed(QString)), this, SLOT(copyPressed(QString)));
+    Q_ASSERT(no_errors);
+    no_errors = connect(rootObject(), SIGNAL(onUploadPressed()), this, SLOT(uploadPressed()));
+    Q_ASSERT(no_errors);
+    no_errors = connect(rootObject(), SIGNAL(onClearPressed()), this, SLOT(clearPressed()));
+    Q_ASSERT(no_errors);
+    no_errors = connect(rootObject(), SIGNAL(onSearchRequested(QString, bool)), this, SLOT(searchRequested(QString, bool)));
+    Q_ASSERT(no_errors);
+
+    loadSettings();
+
+    if (m_model)
+        QMetaObject::invokeMethod(rootObject(), "gotSourceModel");
 }
 
-void LogPage::setInstanceLaunchTaskChanged(shared_qobject_ptr<LaunchTask> proc, bool initial)
+void LogPage::setInstanceLaunchTaskChanged(LaunchTask::Ptr proc, bool initial)
 {
     m_process = proc;
-    if(m_process)
-    {
+    if (m_process) {
         m_model = proc->getLogModel();
         m_proxy->setSourceModel(m_model.get());
-        if(initial)
-        {
-            modelStateToUI();
-        }
-        else
-        {
-            UIToModelState();
-        }
-    }
-    else
-    {
+
+        if (initial)
+            loadSettings();
+
+        QMetaObject::invokeMethod(rootObject(), "gotSourceModel");
+    } else {
         m_proxy->setSourceModel(nullptr);
         m_model.reset();
+
+        QMetaObject::invokeMethod(rootObject(), "lostSourceModel");
     }
 }
 
-void LogPage::onInstanceLaunchTaskChanged(shared_qobject_ptr<LaunchTask> proc)
+void LogPage::onInstanceLaunchTaskChanged(LaunchTask::Ptr proc)
 {
     setInstanceLaunchTaskChanged(proc, false);
 }
 
-bool LogPage::apply()
+void LogPage::wrapModeChanged(int new_state)
 {
-    return true;
+    [[maybe_unused]] bool no_errors = QQmlProperty::write(rootObject(), "wrapMode", new_state);
+    Q_ASSERT(no_errors);
```

Comment:
yeah, a macro that performs that assert without the temporary stack var but retains the readability would be nice